### ISSUE TITLE
What is inside llama-2-70b consolidated.00.pth file and how do I read it?

### ISSUE THREAD

**Issue by imvetri** [facebookresearch/llama#528]
I tried to print out the contents of the `consolidated.00.pth` file using below lines of code. 

```
import torch

model = torch.load('consolidated.00.pth')
print(model)
```

what it struck me was the content after printing like that, the size of the file is around 317kb where as the consolidated.00.pth file is close to 17.25gb.
[weights.txt](https://github.com/facebookresearch/llama/files/12157704/weights.txt)

Are there other contents in the file ? how do I see it? Attached the exported content for reference.

**Comment by michaelnny** [comment]
The content of the .pth file contains the the trained weights and biases of the model. It does not make sense to human eyes.

You need to load it to your model instance, as shown in the example here:

https://github.com/facebookresearch/llama/blob/6c7fe276574e78057f917549435a2554000a876d/llama/generation.py#L83-L97

**Comment by imvetri** [comment]
My Thoughts before this question.

Input + function gives an output is conventional method of computation.m, where as fundamentally AI or ML or DL computes the function with help of input and known outputs.


Ideally, an 100% learnt AI should result in an function, but community seems to reach the state of consciousness or singularity or universe or self awareness system, which was tried before with conventional programs been tried again with modern program which are all a flipped version of the previous paradigm.

In other words, mathematicians tricked human eyes with their theories with symbols and meanings which only they get right, in my words, a job security for mathematicians.

**Comment by XitaoLi** [comment]
> Input + function gives an output is conventional method of computation.m, where as fundamentally AI or ML or DL computes the function with help of input and known outputs.

Good idea. Inspired by your thoughts, i think maybe the next paradigm for DL is training a parameters-adjustable model that fits different functions.

**Comment by imvetri** [comment]
@XitaoLi it will result in Mandelbrot alike recursion

**Comment by hteller22** [comment]
How can I load and use the model ('consolidated.00.pth') and test a prompt? I can do it via GPU or CPU but no idea how to run the whole thing.

**Comment by jboero** [comment]
The .pth files are basically zip files. They should be converted via a convert.py script. See this thread from llama.cpp:

https://github.com/ggerganov/llama.cpp/issues/707

**Comment by KangkangStu** [comment]
Would love to know the difference between these models now, such as bin, pth, pt, onnx, hf, etc. Or how they convert viewing, etc. Because in the process of using it, I found that many models .cpp llama cannot be used in the llama2.c project

**Comment by KangkangStu** [comment]
Would love to know the difference between these models now, such as bin, pth, pt, onnx, hf, etc. Or how they convert viewing, etc. Because in the process of using it, I found that many models uesd in llama.cpp cannot be used in the llama2.c project

**Comment by aelyoussfi** [comment]
same here, the LLM revolutio brings a set of new ways/techs to store and use AI models that I don't know necessarly before. if someone has some ressources to read please share!

**Comment by mustafa404** [comment]
> I tried to print out the contents of the `consolidated.00.pth` file using below lines of code.
> 
> ```
> import torch
> 
> model = torch.load('consolidated.00.pth')
> print(model)
> ```
> 
> what it struck me was the content after printing like that, the size of the file is around 317kb where as the consolidated.00.pth file is close to 17.25gb. [weights.txt](https://github.com/facebookresearch/llama/files/12157704/weights.txt)
> 
> Are there other contents in the file ? how do I see it? Attached the exported content for reference.

How did you extract the weights.txt. I am getting all zeros inside it.
